tests/system/src/test/java/test/javafx/scene/Snapshot1Test.java

```diff
@@ -217,69 +218,76 @@ public void testSceneCallback() {
     @Test
     public void testBadSceneCallback1() {
         final CountDownLatch latch = new CountDownLatch(1);
+        OutputRedirect.suppressStderr();
+        try {
+            Util.runAndWait(() -> {
+                tmpScene = new Scene(new Group(), 200, 100);
 
-        Util.runAndWait(() -> {
-            tmpScene = new Scene(new Group(), 200, 100);
-
-            // NOTE: cannot use a lambda expression for the following callback
-            Callback<SnapshotResult, Void> cb = new Callback() {
-                @Override public Object call(Object param) {
-                    assertNotNull(param);
+                // NOTE: cannot use a lambda expression for the following callback
+                Callback<SnapshotResult, Void> cb = new Callback() {
+                    @Override
+                    public Object call(Object param) {
+                        assertNotNull(param);
+
+                        latch.countDown();
+                        // The following will cause a ClassCastException warning
+                        // message to be printed.
+                        return "";
+                    }
+                };
+
+                tmpScene.snapshot(cb, null);
+                Util.sleep(SLEEP_TIME);
+                assertEquals(1, latch.getCount());
+            });
 
-                    latch.countDown();
-                    // The following will cause a ClassCastException warning
-                    // message to be printed.
-                    return "";
+            try {
+                if (!latch.await(TIMEOUT, TimeUnit.MILLISECONDS)) {
+                    fail("Timeout waiting for snapshot callback");
                 }
-            };
-
-            tmpScene.snapshot(cb, null);
-            Util.sleep(SLEEP_TIME);
-            assertEquals(1, latch.getCount());
-            System.err.println("testBadSceneCallback1: a ClassCastException warning message is expected here");
-        });
-
-        try {
-            if (!latch.await(TIMEOUT, TimeUnit.MILLISECONDS)) {
-                fail("Timeout waiting for snapshot callback");
+            } catch (InterruptedException ex) {
+                fail(ex);
             }
-        } catch (InterruptedException ex) {
-            fail(ex);
-        }
 
-        assertEquals(0, latch.getCount());
+            assertEquals(0, latch.getCount());
+        } finally {
```

Comment:
Since the ClassCast exception happens after the latch countdown, I recommend a sleep (of, say, 100 msec) after checking that the latch count is 0.